includes/class-wc-stripe-order.php

```diff
@@ -0,0 +1,896 @@
+			if ( $amount ) {
+				$this->set_net( $amount );
+			}
+		}
+
+		return $amount;
+	}
+
+	/**
+	 * Updates the Stripe net for order.
+	 *
+	 * @param float  $amount
+	 */
+	public function set_net( $amount = 0.0 ) {
+		$this->update_meta_data( self::META_STRIPE_NET, $amount );
+	}
+
+	/**
+	 * Deletes the Stripe net for order.
+	 */
+	public function delete_net() {
+		$this->delete_meta_data( self::META_STRIPE_NET );
+		$this->delete_meta_data( 'Net Revenue From Stripe' );
+	}
+
+	/**
+	 * Gets the Stripe currency for order.
+	 *
+	 * @return string $currency
+	 */
+	public function get_stripe_currency() {
+		return $this->get_meta( self::META_STRIPE_CURRENCY );
+	}
+
+	/**
+	 * Updates the Stripe currency for order.
+	 *
+	 * @param string $currency
+	 */
+	public function set_stripe_currency( $currency ) {
+		$this->update_meta_data( self::META_STRIPE_CURRENCY, $currency );
+	}
+
+	/**
+	 * Adds metadata to the order to indicate that the payment is awaiting action.
+	 *
+	 * This meta is primarily used to prevent orders from being cancelled by WooCommerce's hold stock settings.
+	 *
+	 * @return void
+	 */
+	public function set_payment_awaiting_action( $save = true ) {
+		$this->update_meta_data( self::META_STRIPE_PAYMENT_AWAITING_ACTION, wc_bool_to_string( true ) );
+
+		if ( $save ) {
+			$this->save();
+		}
+	}
+
+	/**
+	 * Gets the metadata that indicates that the payment is awaiting action.
+	 *
+	 * @return bool Whether the payment is awaiting action.
+	 */
+	public function is_payment_awaiting_action() {
+		return wc_string_to_bool( $this->get_meta( self::META_STRIPE_PAYMENT_AWAITING_ACTION ) );
+	}
+
+	/**
+	 * Removes the metadata from the order that was used to indicate that the payment was awaiting action.
+	 *
+	 * @param bool     $save  Whether to save the order after removing the metadata.
+	 *
+	 * @return void
+	 */
+	public function remove_payment_awaiting_action( $save = true ) {
+		$this->delete_meta_data( self::META_STRIPE_PAYMENT_AWAITING_ACTION );
+
+		if ( $save ) {
+			$this->save();
+		}
+	}
+
+	/**
+	 * Set the preferred card brand.
+	 *
+	 * @param $brand string The brand to set.
+	 * @return void
+	 */
+	public function set_card_brand( $brand ) {
+		$this->update_meta_data( self::META_STRIPE_CARD_BRAND, $brand );
+	}
+
+	/**
+	 * Get the preferred card brand.
+	 *
+	 * @return string
+	 */
+	public function get_card_brand() {
+		return $this->get_meta( self::META_STRIPE_CARD_BRAND );
+	}
+
+	/**
+	 * Locks an order for refund processing for 5 minutes.
+	 *
+	 * @return bool A flag that indicates whether the order is already locked.
+	 */
+	public function lock_refund() {
+		$this->read_meta_data( true );
+
+		$existing_lock = $this->get_lock_refund();
+
+		if ( $existing_lock ) {
+			$expiration = (int) $existing_lock;
+
+			// If the lock is still active, return true.
+			if ( time() <= $expiration ) {
+				return true;
+			}
+		}
+
+		$new_lock = time() + 5 * MINUTE_IN_SECONDS;
```

Comment:
Perhaps we can move it to a variable to avoid the use of magic numbers. 

`$REFUND_LOCK_EXPIRATION = 5 * MINUTE_IN_SECONDS;`

```suggestion
		$new_lock = time() + self::REFUND_LOCK_EXPIRATION;
```